### DISCUSSION TITLE
Clusterization and unsupervised learning

### DISCUSSION THREAD

**Discussion by ghost** [BindsNET/bindsnet#665] [Q&A]
Is it possible to effectively train an LM-SNN neural network by running it on unlabeled data and then labeling the trained and clustered data? This is mentioned in the article, but in the examples provided by BindsNET when training on MNIST, labels of the data used for training via STDP are required for the labeling task. Thus, during the labeling stage, we are dealing with supervised learning.

**Comment by ghost** [comment]
The article mentions the use of self-organizing Kohonen maps. Has this not been implemented in BindsNet yet?

**Reply by Hananel-Hazan** [reply]
Yes you can train `LM-SNN` on unlabeled data and then labeling the trained and clustered data. In fact if you read carefully the `LM-SNN` script, one can see that labeling code in the script is only used for evaluation and doesn't affect training. Therefore, you can safely ignore all the labeling parts and train the network without it. In fact, skipping this part can speed up the training process since it skips the daunting task of testing the accuracy of the network and heavy burden of generating the weights plots that show the spontaneously progress of data clustering formation. 

> The article mentions the use of self-organizing Kohonen maps. Has this not been implemented in BindsNet yet?

I'm confuse, what are the features exist in Kohonen maps that are not found in `LM-SNN`?

**Reply by ghost** [reply]
I thought Kohonen maps were used for clustering after training, perhaps I misunderstood what was written in the article.

**Reply by Hananel-Hazan** [reply]
Right, after the network training finished, both `Kohonen maps` and `LS-SNN` forms spontaneous clusters base on the trained data.
